src/map/itemdb.c

```diff
@@ -1299,6 +1309,90 @@ void itemdb_read_packages(void) {
 	ShowStatus("Done reading '"CL_WHITE"%d"CL_RESET"' entries in '"CL_WHITE"%s"CL_RESET"'.\n", count, config_filename);
 }
 
+/**
+ * Processes any (plugin-defined) additional fields for a itemdb_options entry.
+ *
+ * @param[in,out] entry  The destination ito entry, already initialized
+ *                       (item_opt.index, status.mode are expected to be already set).
+ * @param[in]     t      The libconfig entry.
+ * @param[in]     n      Ordinal number of the entry, to be displayed in case
+ *                       of validation errors.
+ * @param[in]     source Source of the entry (file name), to be displayed in
+ *                       case of validation errors.
+ */
+void itemdb_readdb_options_additional_fields(struct item_option *ito, struct config_setting_t *t, int n, const char *source)
+{
+	// do nothing. plugins can do their own work
+}
+
+/**
+ * Reads the Item Options configuration file.
+ */
+void itemdb_read_options(void)
+{
+	struct config_t item_options_db;
+	struct config_setting_t *ito = NULL, *conf = NULL;
+	int index = 0, count = 0;
+	const char *filepath = "db/item_options.conf";
+	
+	if (!libconfig->load_file(&item_options_db, filepath))
+		return;
+	
+#ifdef ENABLE_CASE_CHECK
+	script->parser_current_file = filepath;
+#endif // ENABLE_CASE_CHECK
+	
+	if ((ito=libconfig->setting_get_member(item_options_db.root, "item_options_db")) == NULL) {
+		ShowError("itemdb_read_options: '%s' could not be loaded.\n", filepath);
+		libconfig->destroy(&item_options_db);
+		return;
+	}
+	
+	while ((conf = libconfig->setting_get_elem(ito, index++))) {
+		struct item_option t_opt = { 0 }, *s_opt = NULL;
+		const char *str = NULL;
+		
+		if (!libconfig->setting_lookup_int16(conf, "Id", &t_opt.index) || t_opt.index <= 0) {
+			ShowError("itemdb_read_options: Invalid Option Id provided for entry %d in '%s', skipping...\n", t_opt.index, filepath);
+			continue;
+		}
+		
+		if (!libconfig->setting_lookup_string(conf, "Name", &str)) {
+			ShowError("itemdb_read_options: Invalid Option variable name provided for Option Id %d in '%s', skipping...\n", t_opt.index, filepath);
```

Comment:
What if it's a duplicated name? And what if it contains invalid characters? This should be handled similarly to the AegisName